src/zenml/config/resource_configuration.py

```diff
@@ -77,7 +77,7 @@ def empty(self) -> bool:
         Returns:
             `True` if no values were configured, `False` otherwise.
         """
-        return len(self.__fields_set__) == 0
+        return len(self.dict(exclude_unset=True, exclude_none=True)) == 0
```

Comment:
What does this do and why does it work? Adding a comment for that might be useful for maintainability